File: Exemplos/App_eXperience_Flutter/android/app/src/main/java/com/elgin/flutter_m8/ServiceSat.java
package com.elgin.flutter_m8;

import android.content.Context;
import android.widget.Toast;

import java.util.Map;
import java.util.Optional;
import java.util.Random;

import br.com.elgin.DeviceInfo;
import br.com.elgin.Sat;
import br.com.elgin.SatInitializer;

public class ServiceSat {
    public static  SatInitializer satInitializer;
    private Context contextSat;

    public ServiceSat(Context context){
        satInitializer = new SatInitializer();
        contextSat = context;
    }

    public String ativarSAT(Map map){
        String retorno = "...";

        int numSessao = (int) map.get("numSessao");
        int subComando = (int) map.get("subComando");
        String codeAtivacao = (String) map.get("codeAtivacao");
        String cnpj = (String) map.get("cnpj");
        int cUF = (int) map.get("cUF");

        retorno = Sat.ativarSat(numSessao, subComando, codeAtivacao, cnpj, cUF);
        mostrarRetorno(retorno);
        return  retorno;
    }

    public String associarAssinatura(Map map) {
        String retorno = "...";

        int numSessao = (int) map.get("numSessao");
        String codeAtivacao = (String) map.get("codeAtivacao");
        String cnpjSh = (String) map.get("cnpjSh");
        String assinaturaAC = (String) map.get("assinaturaAC");

        retorno = Sat.associarAssinatura(numSessao, codeAtivacao, cnpjSh, assinaturaAC);
        mostrarRetorno(retorno);
        return retorno;
    }

    public String consultarSAT(Map map) {
        String result = "...";

        int numSessao = (int) map.get("numSessao");

        result = Sat.consultarSat(numSessao);
        mostrarRetorno(result);
        return result;
    }

    public String statusOperacional(Map map){
        String retorno = "...";

        int numSessao = (int) map.get("numSessao");
        String codeAtivacao = (String) map.get("codeAtivacao");

        retorno = Sat.consultarStatusOperacional(numSessao, codeAtivacao);
        mostrarRetorno(retorno);
        return retorno;
    }

    public String enviarVenda(Map map){
        String retorno = "...";

        int numSessao = (int) map.get("numSessao");
        String codeAtivacao = (String) map.get("codeAtivacao");
        String xmlSale = (String) map.get("xmlSale");

        retorno = Sat.enviarDadosVenda(numSessao, codeAtivacao, xmlSale);
        mostrarRetorno(retorno);
        return retorno;
    }

    public String cancelarVenda(Map map){
        String retorno = "...";

        int numSessao = (int) map.get("numSessao");
        String codeAtivacao = (String) map.get("codeAtivacao");
        String cFeNumber = (String) map.get("cFeNumber");
        String xmlCancelamento = (String) map.get("xmlCancelamento");

        retorno = Sat.cancelarUltimaVenda(numSessao, codeAtivacao, cFeNumber, xmlCancelamento);
        mostrarRetorno(retorno);
        return retorno;
    }


    public String deviceInfo(){
        String retorno = "...";
        Optional<DeviceInfo> deviceInfo = Sat.getDeviceInfo();
        System.out.println(deviceInfo);
        return "aaa";
    }

    public String extrairLog(Map map){
        String retorno = "...";

        int numSessao = (int) map.get("numSessao");
        String codeAtivacao = (String) map.get("codeAtivacao");

        retorno = Sat.extrairLogs(numSessao, codeAtivacao);
        return retorno;
    }


    private void mostrarRetorno(String retorno) {
        Toast.makeText(contextSat, String.format("Retorno: %s", retorno), Toast.LENGTH_LONG).show();
    }
}

File: M8/Exemplos/App_eXperience_Kotlin/app/src/main/java/com/example/e1_elgin_kotlin/ServiceSat.kt
package com.example.e1_elgin_kotlin


import android.content.Context
import android.widget.Toast
import br.com.elgin.Sat
import br.com.elgin.SatInitializer

class ServiceSat(context: Context) {
    private val contextSat: Context
    fun ativarSAT(map: Map<*, *>): String {
        var retorno = "..."
        val numSessao = map["numSessao"] as Int
        val subComando = map["subComando"] as Int
        val codeAtivacao = map["codeAtivacao"] as String?
        val cnpj = map["cnpj"] as String?
        val cUF = map["cUF"] as Int
        retorno = Sat.ativarSat(numSessao, subComando, codeAtivacao, cnpj, cUF)
        mostrarRetorno(retorno)
        return retorno
    }

    fun associarAssinatura(map: Map<*, *>): String {
        var retorno = "..."
        val numSessao = map["numSessao"] as Int
        val codeAtivacao = map["codeAtivacao"] as String?
        val cnpjSh = map["cnpjSh"] as String?
        val assinaturaAC = map["assinaturaAC"] as String?
        retorno = Sat.associarAssinatura(numSessao, codeAtivacao, cnpjSh, assinaturaAC)
        mostrarRetorno(retorno)
        return retorno
    }

    fun consultarSAT(map: Map<*, *>): String {
        var result = "..."
        val numSessao = map["numSessao"] as Int
        result = Sat.consultarSat(numSessao)
        mostrarRetorno(result)
        return result
    }

    fun statusOperacional(map: Map<*, *>): String {
        var retorno = "..."
        val numSessao = map["numSessao"] as Int
        val codeAtivacao = map["codeAtivacao"] as String?
        retorno = Sat.consultarStatusOperacional(numSessao, codeAtivacao)
        mostrarRetorno(retorno)
        return retorno
    }

    fun enviarVenda(map: Map<*, *>): String {
        var retorno = "..."
        val numSessao = map["numSessao"] as Int
        val codeAtivacao = map["codeAtivacao"] as String?
        val xmlSale = map["xmlSale"] as String?
        retorno = Sat.enviarDadosVenda(numSessao, codeAtivacao, xmlSale)
        mostrarRetorno(retorno)
        return retorno
    }

    fun cancelarVenda(map: Map<*, *>): String {
        var retorno = "..."
        val numSessao = map["numSessao"] as Int
        val codeAtivacao = map["codeAtivacao"] as String?
        val cFeNumber = map["cFeNumber"] as String?
        val xmlCancelamento = map["xmlCancelamento"] as String?
        retorno = Sat.cancelarUltimaVenda(numSessao, codeAtivacao, cFeNumber, xmlCancelamento)
        mostrarRetorno(retorno)
        return retorno
    }

    fun deviceInfo(): String {
        val retorno = "..."
        val deviceInfo = Sat.getDeviceInfo()
        return "aaa"
    }

    private fun mostrarRetorno(retorno: String) {
        Toast.makeText(contextSat, String.format("Retorno: %s", retorno), Toast.LENGTH_LONG).show()
    }

    companion object {
        lateinit var satInitializer: SatInitializer
    }

    init {
        satInitializer = SatInitializer()
        contextSat = context
    }
}
